src/Orleans/Messaging/GatewayManager.cs

```diff
@@ -134,6 +134,10 @@ public Uri GetLiveGateway()
             {
                 lock (lockable)
                 {
+                    if (config.PreferedGatewayIndex >= 0 && config.PreferedGatewayIndex < count)
+                    {
+                        return live[config.PreferedGatewayIndex];
```

Comment:
Already talked in person with @benjaminpetit, but we can't do this... He'll have the tests call several times until there's a hide confidence that this path will be hit.